[ru] Localize `function-as-a-service` (#3217)
cncf/glossary: content/ru/function-as-a-service.md

---

> title: Функция-как-сервис (FaaS)

Reviewer: I have the same question regarding hyphens here :sweat_smile: I'd better remove them.

> title: Функция-как-сервис (FaaS)

Author: ```suggestion
title: Функция как сервис (FaaS)
```

status: Completed
category: Technology
tags: ["infrastructure", "", ""]
---

> Функция-как-сервис (Function as a Service, FaaS) — модель облачных вычислений, в которой функции, вызванные событиями, выполняются на сторонней платформе. Ее главный плюс — автоматическое масштабирование, не требующее ручного вмешательства.

Reviewer: ```suggestion
Функция-как-сервис (Function as a Service, FaaS) — модель облачных вычислений, которая предлагает платформу для выполнения функций, инициированных событиями. Она обеспечивает автоматическое масштабирование, не требующее ручного вмешательства.
```

В сущности FaaS позволяет развёртывать отдельные функции, которые активируются в ответ на определенные события, некоторое (короткое) время работают и отключаются. Тем самым гарантируется, что ресурсы не тратятся впустую.

> Модель поддерживает [автоматическое масштабирование](/ru/auto-scaling/), позволяя запускать экземпляр функции по запросу и завершать его после выполнения, подчеркивая stateless-природу.

Reviewer: However, this [original] "stateless" notice doesn't seem to be really helpful here…
```suggestion
Модель поддерживает [автоматическое масштабирование](/ru/auto-scaling/), позволяя запускать экземпляр функции по запросу и завершать его после выполнения, что соответствует его stateless-природе.
```

FaaS-платформы реализуют подход к тарификации по принципу "плати за фактическое использование": когда функция не работает, она не потребляет ресурсы, экономя деньги. Этим они отличаются от других моделей, таких как [Платформа-как-услуга](/platform-as-a-service/) (Platform as a Service, PaaS), которые предполагают постоянную доступность ресурсов.

## Какую проблему решает

Традиционно компании предпочитали работать с собственными центрами обработки данных, что требовало значительных инвестиций в оборудование, программное обеспечение и персонал.

> Такой подход означал, что ЦОД должен был проектироваться под пиковый спрос — т.е. во все остальное время его ресурсы использовались лишь частично.

Reviewer: ```suggestion
Такой подход означал, что ЦОД должен был проектироваться под пиковый спрос, а в остальное время его ресурсы использовались лишь частично.
```

> Кроме того, стремительное развитие бизнеса могло опередить возможности ИТ, приводя к операционой неэффективности.

Reviewer: ```suggestion
Кроме того, стремительное развитие бизнеса могло опередить возможности ИТ и привести к операционной неэффективности.
```

С другой стороны, модели вида [Инфраструктура-как-услуга](/infrastructure-as-a-service/) (Infrastructure-as-a-Service, IaaS), хотя и предлагают облачные решения, все же возлагают бремя масштабирования ресурсов на пользователя, требуя оплаты за постоянную доступность сервера независимо от фактического использования.

## Как именно решает проблему

FaaS предоставляет разработчикам [абстракцию](/ru/abstraction/) для запуска веб-приложений в ответ на события, избавляя их от необходимости управлять серверной инфраструктурой.
Например, загрузка файла может запустить кастомный код, который перекодирует файл в различные форматы.
Инфраструктура FaaS автоматически регулирует ресурсы в зависимости от спроса, освобождая разработчиков от необходимости писать код с учетом [масштабируемости](/scalability/) и связанных с этим сложностей.

> Плата взимается только за время вычислений — когда функции неактивны, деньги не списываются.

Reviewer: ```suggestion
Плата взимается только за время вычислений: когда функции неактивны, деньги не списываются.
```
